roles/ceph-dashboard/tasks/configure_dashboard.yml

```diff
@@ -171,43 +171,69 @@
     - dashboard_protocol == "https"
     - dashboard_grafana_api_no_ssl_verify | bool
 
-- name: set alertmanager host
-  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-alertmanager-api-host http://{{ grafana_server_addrs | first }}:{{ alertmanager_port }}"
-  delegate_to: "{{ groups[mon_group_name][0] }}"
-  run_once: true
-  changed_when: false
+- name: set_fact prometheus_api_host, alertmanager_api_host, grafana_api_url
+  set_fact:
+    prometheus_api_host: "{{ dashboard_protocol }}://{{ prometheus_frontend_vip if prometheus_frontend_vip is defined and prometheus_frontend_vip | length > 0 else monitoring_server_addrs | first }}:{{ prometheus_port }}"
+    alertmanager_api_host: "{{ dashboard_protocol }}://{{ alertmanager_frontend_vip if alertmanager_frontend_vip is defined and alertmanager_frontend_vip | length > 0 else monitoring_server_addrs | first }}:{{ alertmanager_port }}"
```

Comment:
prometheus and alertmanager doesn't support https so you can't reuse the `dashboard_protocol` variable here

Reply:
maggle, this review comes at the right time :grinning: 
:+1:

Reply:
@dsavineau ++ right, prometheus and alertmanager are left w/ http, in OSP for tls-everywhere we're using the approach described in [1] (terminating the ssl on the haproxy frontend vip)

[1] https://github.com/openstack/puppet-tripleo/blob/master/manifests/haproxy.pp#L944-L972